test/e2e/webhook_reserved_namespace_resources_test.go

```diff
@@ -0,0 +1,596 @@
+/*
+Copyright (c) Microsoft Corporation.
+Licensed under the MIT license.
+*/
+
+package e2e
+
+import (
+	"errors"
+	"fmt"
+	"reflect"
+
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	appsv1 "k8s.io/api/apps/v1"
+	batchv1 "k8s.io/api/batch/v1"
+	corev1 "k8s.io/api/core/v1"
+	rbacv1 "k8s.io/api/rbac/v1"
+	k8sErrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/utils/pointer"
+
+	"go.goms.io/fleet/pkg/utils"
+	testutils "go.goms.io/fleet/test/e2e/utils"
+)
+
+const (
+	testRole        = "test-role"
+	testRoleBinding = "test-role-binding"
+	testPod         = "test-pod"
+	testDeployment  = "test-deployment"
+	testJob         = "test-job"
+)
+
+var _ = Describe("Fleet's Reserved Namespaced Resources Handler webhook tests", func() {
```

Comment:
This reminds me of the struct/map style UT we widely use in this repo. In theory, we only need to list all the objects and the tests are the same for each project.